radixdlt-engine/src/main/java/com/radixdlt/constraintmachine/REProcessedTxn.java

```diff
@@ -78,30 +78,11 @@
 import java.util.stream.Stream;
 
 /** Transaction which has been successfully parsed and state checked by radix engine */
-public final class REProcessedTxn {
-  private final List<List<REStateUpdate>> stateUpdates;
-  private final ParsedTxn parsedTxn;
-  private final ECPublicKey signedByKey;
-  private final List<REEvent> events;
-
-  public REProcessedTxn(
-      ParsedTxn parsedTxn,
-      ECPublicKey signedByKey,
-      List<List<REStateUpdate>> stateUpdates,
-      List<REEvent> events) {
-    this.parsedTxn = parsedTxn;
-    this.signedByKey = signedByKey;
-    this.stateUpdates = stateUpdates;
-    this.events = events;
-  }
-
-  public List<REEvent> getEvents() {
-    return events;
-  }
-
-  public ParsedTxn getParsedTxn() {
-    return parsedTxn;
-  }
+public record REProcessedTxn(
+    ParsedTxn parsedTxn,
+    ECPublicKey signedByKey,
+    List<List<REStateUpdate>> stateUpdates,
```

Comment:
Can we call this field `stateUpdateGroups`?

Reply:
Yes. Updated.